gradio/i18n.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+
+class _I18n:
+    """
+    Handles internationalization (i18n) for Gradio applications.
+
+    Stores translation dictionaries and provides a method to retrieve translation keys.
+    The translation lookup happens on the frontend based on the browser's locale
+    and the provided translation dictionaries.
+    """
+
+    def __init__(self, translations: dict[str, dict[str, str]] | None = None):
+        """
+        Initializes the I18n class.
+
+        Args:
+            translations: A dictionary where keys are locale codes (e.g., "en", "es")
+                          and values are dictionaries mapping translation keys to translated strings.
+                          Example: {"en": {"greeting": "Hello"}, "es": {"greeting": "Hola"}}
+        """
+        self.translations = translations or {}
+
+    def __call__(self, key: str) -> str:
```

Comment:
Is  __call__ actually used? My understanding is that the .translations attribute is just passed into the frontend directly